logger_lib/src/logger_lib.hpp

```diff
@@ -0,0 +1,89 @@
+#ifndef LOGGER_LIB_HPP
+#define LOGGER_LIB_HPP
+
+#include <memory>
+#include <string>
+#include <iostream>
+
+#define FLAG_ON(...)  logger::Logging::extract_all_credentials(__DATE__,__TIME__,__FILE__,__FUNCTION__,__LINE__,__VA_ARGS__)
+#define FLAG_OFF(...)	logger::Logging::extract_minimimal_credentials(__DATE__,__TIME__)	
+
+namespace logger
+{
+
+	//Defining log level with severity levels.
+	enum class LEVEL :int { OFF = 0, FATAL = 1, ERROR, WARN, DEBUG, TRACE = 5 };
+	
+	//Logs the credentials.:date, time,file_name,function_name,line_number,logger_name,logger_level,message.
+	//Written in chronological order.	
+	struct  LogCredentials {
+		std::string date="null";
+		std::string time = "null";
+		std::string file_name = "null";
+		std::string function_name = "null";
```

Comment:
this is for over all variable naming, which naming convention are you using?
generally this is used for VB(visual basic), as in VB 
e,g
name and Name considered as same, so to differentiate that they use '_' prefix |  suffix | infix for member variable names
supporting stack overflow answer:
https://stackoverflow.com/questions/3136594/naming-convention-underscore-in-c-and-c-sharp-variables/20496955